data/txHandler.go

```diff
@@ -239,12 +252,39 @@ func (handler *TxHandler) processIncomingTxn(rawmsg network.IncomingMessage) net
 			logging.Base().Warnf("Received a non-decodable txn: %v", err)
 			return network.OutgoingMessage{Action: network.Disconnect}
 		}
+		decTxGroupElemOffsets[ntx] = dec.Consumed()
 		ntx++
+		if ntx >= config.MaxTxGroupSize {
```

Comment:
this is tricky one, and I think it is OK to proceed with this group because:
1. If there is something else then 1) on old nodes it would be an `MaxTxnGroupSize` eval error 2) on new nodes `GroupID` check eval error
2. If there is nothing left then a full group is read and this is fine in old and new nodes.

Reply:
correct, no mspg arrays, just concat and stream